callbarcodes.py

```diff
@@ -0,0 +1,587 @@
+        #
+        # Get the measurements object - we put the measurements we
+        # make in here
+        #
+        measurements = workspace.measurements
+
+        #
+        # We record some statistics which we will display later.
+        # We format them so that Matplotlib can display them in a table.
+        # The first row is a header that tells what the fields are.
+        #
+        statistics = [["Feature", "Mean", "Median", "SD"]]
+
+        #
+        # Put the statistics in the workspace display data so we
+        # can get at them when we display
+        #
+        workspace.display_data.statistics = statistics
+
+        #
+        # Get the input image and object. You need to get the .value
+        # because otherwise you'll get the setting object instead of
+        # the string name.
+        #
+        input_object_name = self.input_object_name.value
+
+        ################################################################
+        #
+        # GETTING AN IMAGE FROM THE IMAGE SET
+        #
+        # Get the image set. The image set has all of the images in it.
+        #
+        image_set = workspace.image_set
+        #
+
+        ###############################################################
+
+        ###############################################################
+        #
+        # GETTING THE LABELS MATRIX FROM THE OBJECT SET
+        #
+        # The object set has all of the objects in it.
+        #
+        object_set = workspace.object_set
+
+        ###############################################################
+
+
+        """
+        for n, m in self.get_zernike_indexes():
+            # Compute the zernikes for each object, returned in an array
+            zr, zi = self.measure_zernike(pixels, labels, indexes, centers, radius, n, m)
+
+            # Get the name of the measurement feature for this zernike
+            feature = self.get_measurement_name(n, m)
+
+            # Add a measurement for this kind of object
+            if m != 0:
+                measurements.add_measurement(input_object_name, feature, zr)
+
+                # Do the same with -m
+                feature = self.get_measurement_name(n, -m)
+                measurements.add_measurement(input_object_name, feature, zi)
+            else:
+                # For zero, the total is the sum of real and imaginary parts
+                measurements.add_measurement(input_object_name, feature, zr + zi)
+
+            # Record the statistics.
+            zmean = numpy.mean(zr)
+            zmedian = numpy.median(zr)
+            zsd = numpy.std(zr)
+            statistics.append([feature, zmean, zmedian, zsd])"""
+
+    #
+    # "display" lets you use matplotlib to display your results.
+    #
+    def display(self, workspace, figure):
+        statistics = workspace.display_data.statistics
+
+        figure.set_subplots((1, 1))
+
+        figure.subplot_table(0, 0, statistics)
+
+
+
+    def likelihood(self,barcode,query):
```

Comment:
It looks like something like [Bio.pairwise2](https://biopython.org/DIST/docs/api/Bio.pairwise2-module.html) could be a promising replacement. Although it does look very similar to the code you have already written here.

Another thing to also consider is if we can implement a more vectorized solution. (i.e. rather than 1 vs. 1 matching, can we do many vs. many?)

I do not know the format of the input or how this module is being used, so this may not be feasible!